packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -764,7 +764,27 @@ export class Project extends ProjectBase {
 
         // save SecretsManager env variables
         if (envVariable.type === BuildEnvironmentVariableType.SECRETS_MANAGER) {
-          if (Token.isUnresolved(envVariableValue)) {
+          if (envVariable.value instanceof Object) {
+            const secretArn = (envVariable.value as secretsmanager.Secret).secretArn;
+            const parsedSecretArn = stack.parseArn(secretArn);
+
+            secretsManagerIamResources.add(`${secretArn}*`);
+            if (parsedSecretArn && parsedSecretArn.account &&
+              Token.compareStrings(stack.account, parsedSecretArn.account) === TokenComparison.DIFFERENT) {
+              kmsIamResources.add(stack.formatArn({
+                service: 'kms',
+                resource: 'key',
+                // We do not know the ID of the key, but since this is a cross-account access,
+                // the key policies have to allow this access, so a wildcard is safe here
+                resourceName: '*',
+                sep: '/',
+                // if we were given an ARN, we need to use the provided partition/account/region
+                partition: parsedSecretArn.partition,
+                account: parsedSecretArn.account,
+                region: parsedSecretArn.region,
```

Comment:
This comment no longer makes sense here, and we don't need `parsedArn`:

```suggestion
                account: secret.env.account,
                region: secreet.env.region,
```